pkgs/development/libraries/libqmi/default.nix

```diff
@@ -42,35 +43,45 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  outputs = [ "out" "dev" ]
+    ++ lib.optional (stdenv.hostPlatform.emulatorAvailable buildPackages) "devdoc";
+
   nativeBuildInputs = [
     meson
     ninja
     pkg-config
-    gobject-introspection
     python3
-  ] ++ lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [
+  ] ++ lib.optionals withMan [
+    help2man
+  ] ++ lib.optionals withIntrospection [
+    gobject-introspection
     gtk-doc
     docbook-xsl-nons
     docbook_xml_dtd_43
-    help2man
+  ] ++ lib.optionals (withIntrospection && !stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+    mesonEmulatorHook
   ];
 
   buildInputs = [
-    libgudev
     bash-completion
     libmbim
+  ] ++ lib.optionals withIntrospection [
+    libgudev
   ];
 
   propagatedBuildInputs = [
     glib
+  ] ++ lib.optionals withIntrospection [
     libqrtr-glib
   ];
 
   mesonFlags = [
     "-Dudevdir=${placeholder "out"}/lib/udev"
-    (lib.mesonBool "gtk_doc" (stdenv.buildPlatform == stdenv.hostPlatform))
-    (lib.mesonBool "introspection" (stdenv.buildPlatform == stdenv.hostPlatform))
-    (lib.mesonBool "man" (stdenv.buildPlatform == stdenv.hostPlatform))
+    (lib.mesonBool "gtk_doc" withIntrospection)
+    (lib.mesonBool "introspection" withIntrospection)
+    (lib.mesonBool "man" withMan)
+    (lib.mesonBool "qrtr" withIntrospection)
+    (lib.mesonBool "udev" withIntrospection)
```

Comment:
i was able to build this with `withIntrospection = false` but `qrtr=true` and `udev=true`, and the resulting binaries launch and link to `libqrtr-glib.so`/`libgudev.so`.

is there a reason this are made conditional on introspection? i admittedly do not know much about libqmi.

Reply:
They are introspection-based libraries, so they won't build for cross setups where we can't do introspection, e.g. s390-linux.